pinot-core/src/main/java/org/apache/pinot/core/operator/transform/function/DateTimeConversionTransformFunction.java

```diff
@@ -93,10 +113,14 @@ public String getName() {
   @Override
   public void init(List<TransformFunction> arguments, Map<String, ColumnContext> columnContextMap) {
     super.init(arguments, columnContextMap);
-    // Check that there are exactly 4 arguments
-    if (arguments.size() != 4) {
-      throw new IllegalArgumentException("Exactly 4 arguments are required for DATE_TIME_CONVERT transform function");
+    String bucketTimeZone = null;
```

Comment:
nit: let's try to use immutable variables. Can you move this null initialization to its own `else` branch in the if below?